src/graph/retrieval.rs

```diff
@@ -105,6 +106,41 @@ where
                 .flatten(),
         ))
     }
+
+    /// Returns a set of all components that match the given predicate, starting
+    /// from the component with the given `component_id`.
+    ///
+    /// If `follow_after_match` is `true`, the search continues deeper beyond
+    /// the matching components.
+    pub(crate) fn find_all(
+        &self,
+        from: u64,
+        mut pred: impl FnMut(&N) -> bool,
+        follow_after_match: bool,
```

Comment:
Minor: I don't understand why it is called `find_all`? 
Because It doesn't finds all nodes. It finds only nodes in direction outgoing. :) 

Maybe `find_with_pred` would be more descriptive?

Reply:
Initially, it also had a direction parameter, but I realized we didn't need it.  I will think of an option.